common/test/java/com/couchbase/lite/internal/core/PeerBindingTest.kt

```diff
@@ -0,0 +1,99 @@
+package com.couchbase.lite.internal.core
+
+
+import com.couchbase.lite.internal.core.peers.TaggedWeakPeerBinding
+import com.couchbase.lite.internal.utils.MathUtils
+import org.junit.Assert
+import org.junit.Test
+import java.lang.IllegalStateException
+
+class PeerBindingTest {
+
+    // Binding valid key value pair should create a binding in map
+    @Test
+    fun testCreateBasicPeerBinding() {
+        val binding = TaggedWeakPeerBinding<Any>()
+        try {
+            val key = binding.reserveKey()
+            val mockObject = object {}
+
+            // reserve a key should create a new entry in the map
+            Assert.assertEquals(1, binding.size())
+            Assert.assertEquals(null, binding.getBinding(key))
+
+            binding.bind(key, mockObject)
+            Assert.assertEquals(1, binding.size())
+            Assert.assertEquals(mockObject, binding.getBinding(key))
+        }
+        finally { binding.clear() }
+    }
+
+    // Trying to bind an object to a nonexistent key leads to exception
+    @Test
+    fun testBindWithoutKey() {
+        val binding = TaggedWeakPeerBinding<Any>()
+        val randomKey = MathUtils.RANDOM.get().nextInt(Int.MAX_VALUE).toLong();
```

Comment:
I'm unclear on what is going on here.  randomKey is definitely not "nonexistent".  It is an integer between 0 and Int.MAX_VALUE-1, inclusive.

Reply:
I need to name that variable differently, it's just a random number, it's not a key added to the map. So that "key" does not exist in the map

Reply:
Oh!!!!  I get it!  You are testing whether you can bind a key that isn't reserved.

Please change the name of the test to reflect that... and as I say below, there is no advantage to using a random number.  All it does is make the test results un-reproducible.  Just pick some number and use it.